probePlugin/src/main/scala/org/virtuslab/ideprobe/handlers/RunConfigurations.scala

```diff
@@ -36,54 +37,81 @@ object RunConfigurations extends IntelliJApi {
+        }
+        case (Some(className), None) => {
+          Option(findPsiClass(className, module)).getOrElse(error(s"Class $className not found"))
+        }
+        case (Some(className), Some(methodName)) => {
+          val psiClass = findPsiClass(className, module)
+          val psiMethods = read {
+            psiClass.getMethods
+          }
+          psiMethods.find(_.getName == methodName)
+            .getOrElse(error(s"Method $methodName not found in class $className. Available methods: ${psiMethods.map(_.getName)}"))
+        }
       }
 
-      dataContext.put(PlatformDataKeys.SELECTED_ITEMS, List(moduleDir).asJava.toArray)
-
-      val wsVirtualFile = VFS.toVirtualFile(projectPath)
-      dataContext.put(PlatformDataKeys.PROJECT_FILE_DIRECTORY, wsVirtualFile)
-
-      val location = PsiLocation.fromPsiElement(moduleDir)
+      val location = read {
+        PsiLocation.fromPsiElement(psiElement)
+      }
       dataContext.put(Location.DATA_KEY, location)
 
       val configurationContext = ConfigurationContext.getFromContext(dataContext)
       val runManager = configurationContext.getRunManager.asInstanceOf[RunManagerEx]
-      val configurationFromContext = runOnUISync(configurationContext.getConfiguration)
+      val configurationFromContext = read {
+        configurationContext.getConfiguration
+      }
 
       runManager.setTemporaryConfiguration(configurationFromContext)
       runManager.setSelectedConfiguration(configurationFromContext)
 
-      val configurations = runOnUISync {
+      val configurations = read {
         configurationContext.getConfigurationsFromContext
       }
-      val producer = runConfiguration.runnerNameFragment match {
+      val producer = runConfiguration.name match {
         case Some(fragment) =>
           configurations
-            .find(_.toString.toLowerCase contains fragment.toLowerCase)
+            .find(_.toString contains fragment)
             .getOrElse(
-              throw new RuntimeException(
+              error(
                 s"Runner name fragment $fragment does not match any configuration name. Available configurations: $configurations."
               )
             )
         case _ =>
           configurations.headOption.getOrElse(
-            throw new RuntimeException(
+            error(
               "No test configuration available for specified settings."
             )
           )
       }
       val selectedConfiguration = producer.getConfigurationSettings
       val transformedConfiguration = RunConfigurationTransformer.transform(selectedConfiguration)
+      transformedConfiguration.getConfiguration match {
+        case config: JUnitConfiguration => {
+          // sets the "use classpath of module:" configuration setting
+          // because it isn't set out of the box in JUnitConfiguration
+          config.getTestObject.getConfiguration.getConfigurationModule.setModule(module)
```

Comment:
I think there is a direct setModule method on JUnitConfiguration

Reply:
It could have been just some local issue because I can't reproduce it anymore